app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -187,27 +178,49 @@ fun MapScreen(
       }
 }
 
+fun inBounds(
+    currentBottomLeft: Point,
+    currentTopRight: Point,
+    loadedBottomLeft: Point,
+    loadedTopRight: Point
+): Boolean {
+  return currentBottomLeft.latitude() >= loadedBottomLeft.latitude() &&
+      currentBottomLeft.longitude() >= loadedBottomLeft.longitude() &&
+      currentTopRight.latitude() <= loadedTopRight.latitude() &&
+      currentTopRight.longitude() <= loadedTopRight.longitude()
+}
+
 /**
- * Get the top right and bottom left coordinates of the current view
+ * Get the bottom left and top right corners of the screen in latitude and longitude coordinates.
+ * The corners are calculated based on the center of the screen and the viewport dimensions. If
+ * useBuffer is true, the corners are calculated with a buffer of 2x the viewport dimensions. This
+ * is useful for loading parkings that are not yet visible on the screen.
  *
- * @param mapView The MapView
- * @return A pair of the top right and bottom left coordinates
+ * @param mapView the MapView to get the screen corners from
+ * @param useBuffer whether to use a buffer to get the corners
+ * @return a pair of the bottom left and top right corners of the screen
  */
-fun getScreenCoordinateFromPoint(mapView: MapView): Pair<Point, Point> {
-
+fun getScreenCorners(mapView: MapView, useBuffer: Boolean = true): Pair<Point, Point> {
```

Comment:
We can make those functions private because they will be useful only for the `MapScreen`

Reply:
done